test/src/test/java/org/corfudb/integration/LogReplicationRoutingQueueIT.java

```diff
@@ -220,6 +223,54 @@ public void testRoutingQueue2way() throws Exception {
         });
     }
 
+    @Test
+    public void testSnapshotCancel() throws Exception {
+
+        // Register client and setup initial group destinations mapping
+        CorfuRuntime clientRuntime = getClientRuntime();
+        CorfuStore clientCorfuStore = new CorfuStore(clientRuntime);
+        String sourceSiteId = DefaultClusterConfig.getSourceClusterIds().get(0);
+        CorfuStoreBrowserEditor editor = new CorfuStoreBrowserEditor(clientRuntime, null, true);
+        String clientName = DEFAULT_ROUTING_QUEUE_CLIENT;
+        RoutingQueueSenderClient queueSenderClient = new RoutingQueueSenderClient(clientCorfuStore, clientName);
+        CountDownLatch startedTransmitting = new CountDownLatch(1);
+        SnapshotProvider snapshotProvider = new SnapshotProvider(clientCorfuStore, startedTransmitting);
+        queueSenderClient.startLRSnapshotTransmitter(snapshotProvider);
+
+        startReplicationServers();
+
+        RoutingQueueListener listener = new RoutingQueueListener(sinkCorfuStores.get(0),
+                sourceSiteId, clientName);
+        sinkCorfuStores.get(0).subscribeRoutingQListener(listener);
+
+        int numFullSyncMsgsGot = listener.snapSyncMsgCnt;
+        boolean sentDuplicateSync = false;
+        boolean cancelledInitialSync = false;
+
+        snapshotProvider.setNumFullSyncBatches(200);
+
+        startedTransmitting.await();
+        log.info("Negotiation snapshot sync has started!!!");
+
+        while (numFullSyncMsgsGot < snapshotProvider.getNumFullSyncBatches()) {
+            while (!cancelledInitialSync) {
+                Map<String, LRFullStateReplicationContext> fullSyncIDs = queueSenderClient.getPendingFullSyncs();
+                if (fullSyncIDs.containsKey(DefaultClusterConfig.getSinkClusterIds().get(0))) {
```

Comment:
As we wait for the latch, it is guaranteed that the map contains the pending full sync for this cluster.  This can be changed to an assert IMO.